### PR TITLE
Add formula for the va_taxable_income variable

### LINKED ISSUE #3310: Add formula for the va_taxable_income variable
Virginia taxable income is currently not defined.

### PR DESCRIPTION
**Pull request by Xinyi-Shao**
Fixes #3310

### Summary
📝➕🧮

This pull request introduces a new variable `va_taxable_income` for the Virginia state income tax calculation. It also updates the `changelog_entry.yaml` file to reflect the minor version increase and the new feature.

> _`va_taxable_income`_
> _New variable for spring tax_
> _Subtracts deductions_

### Walkthrough
* Bump minor version and add changelog entry for new variable ([link](https://github.com/PolicyEngine/policyengine-us/pull/3347/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
* Implement va_taxable_income variable as adjusted gross income minus standard deduction and exemptions ([link](https://github.com/PolicyEngine/policyengine-us/pull/3347/files?diff=unified&w=0#diff-c29cabacdb21583a7d212def5e49d1d66e1bb8911155d5b452f17a44e8e7e658R14-R19))
* Use existing variables and parameters from `policyengine_us` package in formula ([link](https://github.com/PolicyEngine/policyengine-us/pull/3347/files?diff=unified&w=0#diff-c29cabacdb21583a7d212def5e49d1d66e1bb8911155d5b452f17a44e8e7e658R14-R19))

### CODE REVIEW

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py [hunk 1]
```diff
@@ -9,5 +9,16 @@ class va_taxable_income(Variable):
...
         "https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/"
+        "https://www.tax.virginia.gov/sites/default/files/taxforms/individual-income-tax/2022/760-2022.pdf#page=1"
     )
     defined_for = StateCode.VA
+
+    def formula(tax_unit, period, parameters):
+        afagi = tax_unit("va_afagi", period)
+        age_ded = tax_unit("va_age_deduction", period)
+        std_ded = tax_unit("va_standard_deduction", period)
+        itm_ded = tax_unit("va_itemized_deduction", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py:20]:
```suggestion
        itm_ded = tax_unit("va_itemized_deductions", period)
```

File: policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py [hunk 2]
```diff
@@ -9,5 +9,16 @@ class va_taxable_income(Variable):
...
+        vagi = afagi - age_ded
+        ded = itm_ded if itm_ded > 0 else std_ded
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py:22]: cant use if, need to use `where`

File: policyengine_us/variables/gov/states/va/tax/income/va_agi.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from policyengine_us.model_api import *
+
+
+class va_agi(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Virginia adjusted federal adjusted gross income (VAGI)"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/va/tax/income/va_agi.py:7]:
```suggestion
    label = "Virginia adjusted federal adjusted gross income"
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py [hunk 4]
```diff
@@ -9,5 +9,16 @@ class va_taxable_income(Variable):
...
+        "https://www.tax.virginia.gov/sites/default/files/taxforms/individual-income-tax/2022/760-2022.pdf#page=1"
     )
     defined_for = StateCode.VA
+
+    def formula(tax_unit, period, parameters):
+        agi = tax_unit("va_agi", period)
+        age_ded = tax_unit("va_age_deduction", period)
+        std_ded = tax_unit("va_standard_deduction", period)
+        itm_ded = tax_unit("va_itemized_deductions", period)
+        vagi = agi - age_ded
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py:21]: VAGI = virginia adjusted gross income

image

File: policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py [hunk 5]
```diff
@@ -9,5 +9,17 @@ class va_taxable_income(Variable):
...
     )
     defined_for = StateCode.VA
+
+    def formula(tax_unit, period, parameters):
+        agi = tax_unit("va_agi", period)
+        age_ded = tax_unit("va_age_deduction", period)
+        std_ded = tax_unit("va_standard_deduction", period)
+        itm_ded = tax_unit("va_itemized_deductions", period)
+        itemizes = tax_unit("tax_unit_itemizes", period)
+        agi_less_age_ded = agi - age_ded
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py:22]:
```suggestion
```

File: policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py [hunk 6]
```diff
@@ -9,5 +9,17 @@ class va_taxable_income(Variable):
...
+        ded = where(itemizes, itm_ded, std_ded)
+        exemptions = tax_unit("va_total_exemptions", period)
+        return max_(agi_less_age_ded - ded - exemptions, 0)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py:25]:
```suggestion
        total_deductions = age_ded + ded + exemptions
        return max_(agi - total_deductions, 0)
```

**Review by MaxGhenis — APPROVED** [review]

File: policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/va/tax/income/va_taxable_income.py]: thanks for clarifying the python variables

**Pull request merged**